p2p: Disconnect, not discourage, misbehaving NODE_BLOOM peers

maflcko opened the pull request:
This continues the story started in commit 3a10d935ac8ebabdfd336569d943f042ff84b13e

Xekyo commented on src/net_processing.cpp:
```diff
@@ -3709,20 +3709,16 @@ void PeerManager::ProcessMessage(CNode& pfrom, const std::string& msg_type, CDat
                 pfrom.m_tx_relay->pfilter->insert(vData);
             } else {
-                bad = true;
+                pfrom.fDisconnect = true;
```
This behavior seems to have previously triggered log messages "too-large bloom filter` or  `bad filteradd message`. I see that in other instances of `pfrom.fDisconect` it seems to be often preceded by log messages, e.g. `LogPrintf("Disconnecting outbound peer %d -- headers chain has insufficient work\n", pfrom.GetId());`.

Should there be a log message here for the disconnect as well?

Commit:
p2p: Disconnect, not discourage, misbehaving NODE_BLOOM peers

Receiving an oversized FILERLOAD or FILTERADD message is about as costly
as receiving any large-sized unknown message. Light clients are already
preferred for eviction, so no need to additionally discourage them.

jnewbery commented:
Can you add a motivation to the PR description please? :)

jnewbery reviewed (commented):
I'm not sure if this is an improvement. Setting `fDisconnect` directly bypasses the logic in `MaybeDiscourageAndDisconnect()` that prevents us from disconnecting peers with `PF_NOBAN` or manual connections. Is that intentional?

MarcoFalke commented:
Sorry, this was opened accidentally (not ready yet)

jnewbery commented on src/net_processing.cpp:
```diff
@@ -3709,20 +3709,16 @@ void PeerManager::ProcessMessage(CNode& pfrom, const std::string& msg_type, CDat
         if (vData.size() > MAX_SCRIPT_ELEMENT_SIZE) {
-            bad = true;
+            pfrom.fDisconnect = true;
         } else if (pfrom.m_tx_relay != nullptr) {
```
Actually this is impossible, since we'll never advertise `NODE_BLOOM` for outbound peers.
